src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -6308,71 +6308,131 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
         noway_assert(uCntBytes == 0);
 
 #elif defined(TARGET_XARCH)
-        /*
-            Generate the following code:
-
-                lea     edi, [ebp/esp-OFFS]
-                mov     ecx, <size>
-                xor     eax, eax
-                rep     stosd
-         */
+        noway_assert(compiler->getSIMDSupportLevel() >= SIMD_SSE2_Supported);
+        // As we output multiple instructions for SIMD zeroing, we want to balance code size, with throughput
+        // so cap max size at 16 * MAX SIMD length
+        int initMaxSIMDSize = 16 * XMM_REGSIZE_BYTES;
+        if ((untrLclHi - untrLclLo) <= initMaxSIMDSize)
+        {
+            /*
+                Generate the following code:
+                   xor      ecx, ecx
```

Comment:
Nit: Typically, it is not going to be ecx. genGetZeroReg tries to avoid collisions with the enregistered arguments.

Reply:
Looks to always use `rax`

Reply:
Updated